skill-tree/src/main/java/com/RDS/skilltree/services/AuthServiceImplementation.java

```diff
@@ -0,0 +1,41 @@
+package com.RDS.skilltree.services;
+
+import com.RDS.skilltree.services.external.RdsService;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthServiceImplementation implements AuthService {
+    private final RdsService rdsService;
+
+    @Value("${skilltree.frontendBaseUrl}")
+    private String skilltreeFrontendBaseUrl;
+
+    @Value("${skilltree.backendBaseUrl}")
+    private String skilltreeBackendBaseUrl;
+
+    @Override
+    public void signInWithRds(HttpServletRequest request, HttpServletResponse response)
+            throws IOException {
+        String callbackUrl = skilltreeBackendBaseUrl + "/v1/auth/rds/callback";
+        String redirectUrl = rdsService.signIn(callbackUrl);
+
+        response.sendRedirect(redirectUrl);
+    }
+
+    @Override
+    public void signInWithRdsCallback(HttpServletRequest request, HttpServletResponse response)
```

Comment:
here critical issue is that the code directly echoes the authentication cookie back to the client in the Set-Cookie header. This is a severe security vulnerability as it exposes sensitive information.

Never echo the authentication cookie back to the client. It should be generated securely on the server and sent only once to the client.

```
@Override
public void signInWithRdsCallback(HttpServletRequest request, HttpServletResponse response)
        throws IOException {
    // Generate a secure authentication token or session ID
    String authToken = generateSecureAuthToken();

    // Set the authentication token as a secure cookie
    Cookie cookie = new Cookie("auth_token", authToken);
    cookie.setHttpOnly(true); // Prevent client-side JavaScript access
    cookie.setSecure(true); // Only transmit over HTTPS
    response.addCookie(cookie);

    // Redirect to the dashboard
    String dashboardUrl = skilltreeFrontendBaseUrl + "/dashboard";
    response.sendRedirect(dashboardUrl);
}

```